app/javascript/react/components/MoviesFormContainer.js

```diff
@@ -0,0 +1,126 @@
+import React, { useState } from 'react'
+import { Redirect } from 'react-router-dom'
+import _ from 'lodash'
+
+import ErrorList from './ErrorList'
+
+const MoviesFormContainer = (props) => {
+  const [submittedMovie, setSubmittedMovie] = useState({
+    title: "",
+    summary: "",
+    year: ""
+  })
+  const [shouldRedirect, setShouldRedirect] = useState({
+    redirect: false,
+    id: ""
+  })
+  const [errors, setErrors] = useState({})
+
+  const validforSubmission = () => {
+    let submittedErrors = {}
+    const requiredFields = ["title", "summary", "year"]
+    requiredFields.forEach(field => {
+      if (submittedMovie[field].trim() === "") {
+        submittedErrors = {
+          ...submittedErrors,
+          [field]: "is blank"
+        }
+      }
+    })
+    setErrors(submittedErrors)
+    return _.isEmpty(submittedErrors)
+  }
+
+  const inputChangeHandler = (event) => {
+    setSubmittedMovie({
+      ...submittedMovie,
+      [event.currentTarget.name]: event.currentTarget.value
+    })
+  }
+
+  const onClickHandler = (event) => {
+    event.preventDefault()
+    if (validforSubmission()) {
+      fetch('/api/v1/movies.json', {
+        method: "POST",
+        body: JSON.stringify(submittedMovie),
+        credentials: "same-origin",
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json'
+        }
+      })
+      .then(response => {
+        if (response.ok) {
+          return response.json()
+        } else {
+          // error stuff
```

Comment:
We need _real_ error stuff here with a catch statement!

Reply:
added!